codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -1851,4 +2029,332 @@ type TestEntity {
+            }
+        }
+
+        // Now use the helper function to test all fields
+
+        // BigInt fields
+        check_field_type(
+            entity.fields.get("exampleBigInt").expect("Field not found"),
+            "BigInt",
+            Some(76),
+            None,
+            false, // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigIntRequired")
+                .expect("Field not found"),
+            "BigInt",
+            Some(77),
+            None,
+            true,  // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigIntArray")
+                .expect("Field not found"),
+            "BigInt",
+            Some(78),
+            None,
+            false, // is_required
+            true,  // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigIntArrayRequired")
+                .expect("Field not found"),
+            "BigInt",
+            Some(79),
+            None,
+            true, // is_required
+            true, // is_array
+        );
+
+        // BigDecimal fields
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimal")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(80),
+            Some(5),
+            false, // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalRequired")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(81),
+            Some(5),
+            true,  // is_required
+            false, // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalArray")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(82),
+            Some(5),
+            false, // is_required
+            true,  // is_array
+        );
+
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalArrayRequired")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(83),
+            Some(5),
+            true, // is_required
+            true, // is_array
+        );
+
+        // exampleBigDecimalOtherOrder
+        check_field_type(
+            entity
+                .fields
+                .get("exampleBigDecimalOtherOrder")
+                .expect("Field not found"),
+            "BigDecimal",
+            Some(84),
+            Some(6),
+            true,  // is_required
+            false, // is_array
+        );
+    }
+
+    #[test]
```

Comment:
The next five tests test different failure cases and make sure that a nice easy to read error message is printed for users so they can easily understand their mistake.